lfs/ntlm.go

```diff
@@ -169,13 +171,147 @@ func cloneRequestBody(req *http.Request) (io.ReadCloser, error) {
 		return nil, nil
 	}
 
-	buf, err := ioutil.ReadAll(req.Body)
+	var cb *cloneableBody
+	var err error
+	hasExisting := false
+
+	// check to see if the request body is already a cloneableBody
+	body := req.Body
+	if existingCb, ok := body.(*cloneableBody); ok {
+		hasExisting = true
+		cb, err = existingCb.CloneBody()
+	} else {
+		cb, err = newCloneableBody(req.Body, 0)
+	}
+
 	if err != nil {
 		return nil, err
 	}
 
-	req.Body = ioutil.NopCloser(bytes.NewBuffer(buf))
-	return ioutil.NopCloser(bytes.NewBuffer(buf)), nil
+	// if the request body is NOT a cloneable body, replace it with one
+	if !hasExisting {
+		cb2, err := cb.CloneBody()
+		if err != nil {
+			return nil, err
+		}
+
+		req.Body = cb2
+	}
+
+	return cb, nil
+}
+
+type cloneableBody struct {
+	b      []byte    // in-memory buffer of body
```

Comment:
Some internal go code uses short vars like this. The go community seems to favor shorter names like that. But I think I learn more towards your opinion on this :)